relay-config/src/config.rs

```diff
@@ -1044,7 +1067,17 @@ impl Config {
 
     /// Returns the emit_outcomes flag
     pub fn emit_outcomes(&self) -> bool {
-        self.values.relay.emit_outcomes
+        self.values.outcomes.emit_outcomes
```

Comment:
_nit_: What if you also check `processing_enabled` here, so that there is no chance that the caller forgets this?

In the end the outcomes actor still has to check for processing to know how to send the outcome, but maybe we will have some other places where this becomes relevant.

Reply:
Not 100% sure about what are you suggesting. 

Currently, if processing is enabled, `emit_outcomes` is irrelevant ( that is outcomes are sent to kafka regardless of the flag). This was done to maintain backward compatibility, since emit_outcomes is by default disabled checking for emit_outcomes in processing mode would change the behaviour of existing configurations ( they would not send anything unless the emit_outcomes configuration flag was added).

So current behaviour is:
`processing_enabled == true`--> outcomes go to kafka topic (regardless of emit_outcomes)
'processing_enabled == false` --> outcomes go to http `if emit_outcomes == true`

**Do we want to change this ?**

Reply:
Right, we shouldn't change the behavior.

Let's say in the future we add some log output in startup that says: "outcome logging is enabled", then it would be nice to just call `config.emit_outcomes()` rather than two functions. 

Feel free to disregard this comment if you think that it makes things intransparent.

Reply:
At the moment if `processing_enabled == true` we always emit outcomes regardless of  the `emit_outcomes` flag.
This is done mostly for backward compatibility, here's the reason:

We want an external user not to bother with setting up this flag, and therefore we want to default it to `emit_outcomes == false`

At the same time we want to be backward compatible so an old configuration for a processing relay functions as before:

An old configuration would not have the flag `emit_outcomes` and therefore would have the default `emit_outcomes==false` but it would have `processing==true`.

In order to both maintain compatibility with old configurations for internal relays and offer good defaults for external users we have `emit_outcomes` flag ignored when `processing==true`.

** Is this what we want or do we want to change this behaviour ?**